spinalcordtoolbox/reports/qc.py

```diff
@@ -422,7 +424,8 @@ def update_description_file(self, dimension):
         if os.path.isfile(self.qc_params.qc_results):
             results = json.load(open(self.qc_params.qc_results, 'r'))
```

Comment:
we should probably lock the file and **then** read it (instead of locking it after), otherwise content might change between this line and the line below where you write

Reply:
@npgoodhue you can open with 'r+', load, then `.seek(0)` and dump.

Reply:
My fault, I tried to give hints but didn't think this through.